### ISSUE TITLE
False positive with PLE4703 and immediate return / break

### ISSUE THREAD

**Issue by hauntsaninja** [astral-sh/ruff#10721]
Not the biggest deal, but I got a false positive on this with:
```
λ cat x.py
def foo():
    s = {1, 2, 3}
    for e in s:
        if e >= 3:
            s.remove(e)
            return

λ ruff check --select PLE x.py --preview 
x.py:3:5: PLE4703 Iterated set `s` is modified within the `for` loop
  |
1 |   def foo():
2 |       s = {1, 2, 3}
3 |       for e in s:
  |  _____^
4 | |         if e >= 3:
5 | |             s.remove(e)
6 | |             return
  | |__________________^ PLE4703
  |
  = help: Iterate over a copy of `s`

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

cc @boolean-light in case you're interested, and thanks for the new rule :-)

**Comment by charliermarsh** [comment]
Thanks!

**Comment by charliermarsh** [comment]
Struggling not to get sniped into fixing this.

**Comment by MichaReiser** [comment]
Hmm that makes sense. It would be nice to use the control flow graph for this because I assume there are other situations where the loop exits early (`continue`, `break`, `raise` etc) that aren't covered and they might be more subtle.

**Comment by boolean-light** [comment]
Excuse me I'm back :D Thank you, and let me have a look on this.

**Comment by boolean-light** [comment]
I'm working on this right now, should I exclude warning for unreachable code like this?
```python
s = {1, 2, 3}
for i in s:
    break
    s.discard(i)
```

**Comment by boolean-light** [comment]
Sorry, actually I don't think I can implement this one by myself. Is there some kind of support for control flow group in Ruff already?

**Comment by MichaReiser** [comment]
No, there's not. That's what I meant by my comment. We probably want to wait for a more complete control flow analysis or decided to only support a very small subset of control flow patterns for now.
